desktop/libs/notebook/src/notebook/connectors/sql_alchemy.py

```diff
@@ -211,6 +214,20 @@ def _get_session(self, notebook, snippet):
 
     return None
 
+
+  def _create_connection(self, engine):
+    connection = None
+    try:
+      connection = engine.connect()
+    except Exception as e:
+      engine_key = self._get_engine_key()
+      del ENGINES[engine_key]
+      
+      raise AuthenticationRequired(message='Could not establish connection to datasource')
```

Comment:
And if in the long term we need to bubble up to the user more troubleshooting info to the end user, we could provide/log the message from the `e` exception

Reply:
Yes something like so:

```python
message = force_unicode(e)
raise AuthenticationRequired(message=message)
```

Tried that initially but it's weird SQLAlchemy gives a blank message when the credentials to the datasource is invalid. Not sure if the same behaviour will occur with other datasources other than Databricks; the datasource used in manually testing this change

Reply:
:+1: , let's go without and see if we need to improve it